src/constants/engine-state.ts

```diff
@@ -0,0 +1,14 @@
+enum STATES {
```

Comment:
Mb `EngineState`?
It also could be used as a value type.
Also, we already use `EngineState` like names in other enums.